tests/common/member_test.go

```diff
@@ -52,3 +55,157 @@ func TestMemberList(t *testing.T) {
 		})
 	}
 }
+
+func TestMemberAdd(t *testing.T) {
+	testRunner.BeforeTest(t)
+
+	tcs := []struct {
+		name    string
+		learner bool
+	}{
+		{
+			name:    "NotLearner",
+			learner: false,
+		},
+		{
+			name:    "Learner",
+			learner: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		for _, nc := range clusterTestCases {
+			t.Run(tc.name+"/"+nc.name, func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+				clus := testRunner.NewCluster(ctx, t, nc.config)
+				defer clus.Close()
+				cc := clus.Client()
+
+				testutils.ExecuteUntil(ctx, t, func() {
+					clus.WaitLeader(t)
+					var addResp *clientv3.MemberAddResponse
+					var err error
+					if tc.learner {
+						addResp, err = cc.MemberAddAsLearner(ctx, "newmember", []string{"http://localhost:123"})
+					} else {
+						addResp, err = cc.MemberAdd(ctx, "newmember", []string{"http://localhost:123"})
+					}
+					if nc.config.ClusterSize == 1 {
+						checkAddResp(t, addResp, err)
```

Comment:
I would expect reverse result, adding invalid (`localhost:123` will never respond) member should fail in single node cluster as you cannot establish quorum. From 2 members only 1 is alive, but quorum is 2.

Reply:
Actually, adding an invalid member succeeds in the original e2e test. I also tried this in my terminal, it succeeded too. Is this a long-standing bug?🤣